build.yaml

```diff
@@ -2896,3 +2858,88 @@ steps:
     - deploy_batch
     - deploy_shuffler
     - deploy_address
+ - kind: runImage
+   name: test_batch_invariants
+   image:
+     valueFrom: batch_image.image
+   script: |
+     python3 -m pytest --log-date-format="%Y-%m-%dT%H:%M:%S" --log-format="%(asctime)s %(levelname)s %(name)s %(filename)s:%(lineno)d:%(funcName)s %(message)s" --log-cli-level=INFO -s -vv --instafail -k "test_invariants" /io/test/
+   inputs:
+    - from: /repo/batch/test
+      to: /io/
+   timeout: 300
+   secrets:
+    - name: gce-deploy-config
+      namespace:
+        valueFrom: default_ns.name
+      mountPath: /deploy-config
+    - name: test-dev-tokens
+      namespace:
+        valueFrom: batch_pods_ns.name
+      mountPath: /user-tokens
+    - name: ssl-config-batch-tests
+      namespace:
+        valueFrom: default_ns.name
+      mountPath: /ssl-config
+   scopes:
+    - test
+    - dev
+   dependsOn:
+    - create_deploy_config
+    - create_accounts
+    - default_ns
+    - batch_pods_ns
+    - copy_files
+    - batch_image
+    - deploy_batch
+    - test_batch_0
+    - test_batch_1
+    - test_batch_2
+    - test_batch_3
+    - test_batch_4
+    - test_ci
+    - test_hailtop_batch_0
+    - test_hailtop_batch_1
+    - test_hailtop_batch_2
+    - test_hailtop_batch_3
+    - test_hailtop_batch_4
+    - test_query
+ - kind: runImage
+   name: delete_batch_instances
+   image:
+     valueFrom: base_image.image
+   alwaysRun: true
+   script: |
+     set -ex
+     gcloud -q auth activate-service-account --key-file=/test-gsa-key/key.json
+     set +e
+     gcloud -q compute instances list \
+         --filter 'tags.items=batch2-agent AND labels.namespace={{ default_ns.name }}' \
+         --format="table[no-heading](zone.basename(), name)" \
+         --project {{ global.project }} \
+       | xargs -n2 -r sh -c 'gcloud -q compute instances delete --zone "$1" --project {{ global.project }} "$2" || true' argv0
+   secrets:
+    - name: test-gsa-key
+      namespace:
+        valueFrom: batch_pods_ns.name
+      mountPath: /test-gsa-key
+   scopes:
+    - dev
+    - test
+   dependsOn:
+    - default_ns
+    - batch_pods_ns
+    - base_image
+    - test_batch_invariants
```

Comment:
What happens if you depend on a step that isn't in the deploy scope? Are you just ignored?

Reply:
Yeah the `job` is `None` so `wrapped_job` returns the empty list, so it is effectively not a dependency for its dependents.